Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -3432,37 +3432,46 @@ internal bool RemoveTransferredSubscription(Subscription subscription)
-
-                ClientBase.ValidateResponse(results, subscriptionIds);
-                ClientBase.ValidateDiagnosticInfos(diagnosticInfos, subscriptionIds);
+                    ResponseHeader responseHeader = TransferSubscriptions(null, subscriptionIds, sendInitialValues, out var results, out var diagnosticInfos);
+                    if (!StatusCode.IsGood(responseHeader.ServiceResult))
+                    {
+                        Utils.LogError("TransferSubscription failed: {0}", responseHeader.ServiceResult);
+                        return false;
+                    }
+                    ClientBase.ValidateResponse(results, subscriptionIds);
+                    ClientBase.ValidateDiagnosticInfos(diagnosticInfos, subscriptionIds);
 
-                for (int ii = 0; ii < subscriptions.Count; ii++)
-                {
-                    if (StatusCode.IsGood(results[ii].StatusCode))
+                    for (int ii = 0; ii < subscriptions.Count; ii++)
                     {
-                        if (subscriptions[ii].Transfer(this, subscriptionIds[ii], results[ii].AvailableSequenceNumbers))
-                        {   // create ack for available sequence numbers
-                            foreach (var sequenceNumber in results[ii].AvailableSequenceNumbers)
-                            {
-                                var ack = new SubscriptionAcknowledgement() {
-                                    SubscriptionId = subscriptionIds[ii],
-                                    SequenceNumber = sequenceNumber
-                                };
-                                m_acknowledgementsToSend.Add(ack);
+                        if (StatusCode.IsGood(results[ii].StatusCode))
+                        {
+                            if (subscriptions[ii].Transfer(this, subscriptionIds[ii], results[ii].AvailableSequenceNumbers))
+                            {   // create ack for available sequence numbers
+                                foreach (var sequenceNumber in results[ii].AvailableSequenceNumbers)
+                                {
+                                    var ack = new SubscriptionAcknowledgement() {
+                                        SubscriptionId = subscriptionIds[ii],
+                                        SequenceNumber = sequenceNumber
+                                    };
+                                    m_acknowledgementsToSend.Add(ack);
+                                }
                             }
                         }
-                    }
-                    else
-                    {
-                        Utils.LogError("SubscriptionId {0} failed to transfer, StatusCode={1}", subscriptionIds[ii], results[ii].StatusCode);
+                        else
+                        {
+                            Utils.LogError("SubscriptionId {0} failed to transfer, StatusCode={1}", subscriptionIds[ii], results[ii].StatusCode);
+                            return false;
```

Comment:
The loop here should not exit early, finish all subscriptions in the loop and store result here, then return the false result.

Reply:
Hello @mregen .  I already updated the log info and now the loop will not exit early. The reason I exit early in previous code is I think if any of the subscriptions transfer failed(should happened for the first transfer trial), all subscriptions should be re-created. 

If there is any other logic error, please tell me. Thanks.